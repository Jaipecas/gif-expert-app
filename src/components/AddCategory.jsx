import PropTypes from "prop-types";
import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setinputValue] = useState("");

  //{target} sirve para desesctrucutar las propuiedades del evento y quedarnos con la que nos interesa
  const onInputChange = ({ target }) => {
    setinputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();

    if (inputValue.trim().length === 0) return;

    onNewCategory(inputValue.trim());
    setinputValue("");
  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar gifd"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
