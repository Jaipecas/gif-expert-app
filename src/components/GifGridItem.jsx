export const GifGridItem = ({ title, url }) => {
  return (
    <div className="card">
      <img src={url} />
      <p>{title}</p>
    </div>
  );
};
