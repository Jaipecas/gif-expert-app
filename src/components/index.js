//Sirve para organizar las importaciones, si se apunta a la carpeta componentes y no se especifica nada
//por defecto va al archivo index
//se llaman archivos de barril

export * from "./AddCategory";
export * from "./GifGrid";
export * from "./GifGridItem";
