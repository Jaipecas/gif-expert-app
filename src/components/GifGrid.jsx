import PropTypes from "prop-types";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);
  return (
    <>
      <h2>{category}</h2>
      {
        isLoading && <h2>Cargando...</h2>
      }
      <div className="card-grid">
        {images.map((image) => (
          //el codigo {...image} permite pasar todas las prpiedades del objeto como props al componente hijo
          <GifGridItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
