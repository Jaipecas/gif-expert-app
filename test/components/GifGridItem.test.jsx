//intentar crear el la carpeta de test la misma estructura de carpetas que tenemos en src

import { GifGridItem } from "../../src/components/GifGridItem";
import { render, screen } from "@testing-library/react";

describe("Pruebas gifGridItem", () => {
  const titleTest = "Test";
  const url = "https://loquesea.api/";

  test("should ", () => {
    //del render obtienes el container es decir el elemento div del componente
    const { container } = render(<GifGridItem title={titleTest} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("should ", () => {
    render(<GifGridItem title={titleTest} url={url} />);
    //genera in console.log del componente renderizado
    //screen.debug();

    //se obtiene el elemento html que se quiere testear
    const imgElement = screen.getByRole("img");

    expect(imgElement.src).toBe(url);
    expect(imgElement.alt).toBe(titleTest);
  });

  test("should que exista un texto en el componente", () => {
    render(<GifGridItem title={titleTest} url={url} />);

    expect(screen.getByText("Test")).toBeTruthy();
  });
});
