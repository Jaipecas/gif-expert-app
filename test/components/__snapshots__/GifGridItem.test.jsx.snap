// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas gifGridItem should  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test"
      src="https://loquesea.api/"
    />
    <p>
      Test
    </p>
  </div>
</div>
`;
